src/mina-signer/tests/zkapp.unit-test.ts

```diff
@@ -0,0 +1,113 @@
+import { ZkappCommand } from '../../provable/gen/transaction-bigint.js';
+import * as TransactionJson from '../../provable/gen/transaction-json.js';
+import Client from '../MinaSigner.js';
+import { accountUpdateExample } from '../src/test-vectors/accountUpdate.js';
+import { expect } from 'expect';
+import { Transaction } from '../../lib/mina.js';
+import { PrivateKey } from '../../lib/signature.js';
+import { isReady, shutdown } from '../../snarky.js';
+import { Signature } from '../src/signature.js';
+
+const client = new Client({ network: 'testnet' });
+let { publicKey, privateKey } = client.genKeys();
+
+let dummy = ZkappCommand.toJSON(ZkappCommand.emptyValue());
+let dummySignature = Signature.toBase58(Signature.dummy());
+
+// we construct a transaction which needs signing of the fee payer and another account update
+let accountUpdateExample2: TransactionJson.AccountUpdate = {
+  ...accountUpdateExample,
+  body: {
+    ...accountUpdateExample.body,
+    publicKey,
+    authorizationKind: 'Signature',
+  },
+  authorization: { proof: null, signature: dummySignature },
+};
+
+let exampleZkappCommand: TransactionJson.ZkappCommand = {
+  ...dummy,
+  accountUpdates: [accountUpdateExample, accountUpdateExample2],
+  memo: 'E4YM2vTHhWEg66xpj52JErHUBU4pZ1yageL4TVDDpTTSsv8mK6YaH',
+};
+
+let exampleFeePayer = {
+  feePayer: publicKey,
+  fee: '100000000',
+  nonce: '1',
+  memo: 'test memo',
+};
+
+// generates and verifies a signed zkapp command
+let zkappCommand = client.signZkappCommand(
+  { zkappCommand: exampleZkappCommand, feePayer: exampleFeePayer },
+  privateKey
+);
+expect(zkappCommand.data).toBeDefined();
+expect(zkappCommand.signature).toBeDefined();
+expect(client.verifyZkappCommand(zkappCommand)).toEqual(true);
+expect(client.verifyTransaction(zkappCommand)).toEqual(true);
+
+// generates and verifies a signed zkapp command by using signTransaction
+zkappCommand = client.signTransaction(
+  { zkappCommand: exampleZkappCommand, feePayer: exampleFeePayer },
+  privateKey
+);
+expect(zkappCommand.data).toBeDefined();
+expect(zkappCommand.signature).toBeDefined();
+expect(client.verifyZkappCommand(zkappCommand)).toEqual(true);
+expect(client.verifyTransaction(zkappCommand)).toEqual(true);
+
+// does not verify a signed zkapp command from `mainnet`
+const mainnetClient = new Client({ network: 'mainnet' });
+expect(mainnetClient.verifyZkappCommand(zkappCommand)).toEqual(false);
+expect(mainnetClient.verifyTransaction(zkappCommand)).toEqual(false);
+
+// should throw an error if no fee is passed to the fee payer
+expect(() => {
+  client.signZkappCommand(
+    {
+      zkappCommand: exampleZkappCommand,
+      // @ts-ignore - fee is not defined
+      feePayer: { feePayer: publicKey, nonce: '0', memo: 'test memo' },
+    },
+    privateKey
+  );
+}).toThrow('Fee must be greater than');
```

Comment:
nit to the error that `signZkappCommand` throws: I think it would be good to explicitly differentiate between the two cases - no fee specified and fee not high enough